Common/gfx/allegrobitmap.cpp

```diff
@@ -137,31 +139,18 @@ bool Bitmap::LoadFromFile(const char *filename)
 {
     Destroy();
 
-	BITMAP *al_bmp = load_bitmap(filename, nullptr);
-	if (al_bmp)
-	{
-		_alBitmap = al_bmp;
-        _isDataOwner = true;
-	}
-	return _alBitmap != nullptr;
-}
-
-bool Bitmap::LoadFromFile(PACKFILE *pf)
-{
-    Destroy();
+    std::unique_ptr<Bitmap> bmp (BitmapHelper::LoadFromFile(filename));
 
-    BITMAP *al_bmp = load_bmp_pf(pf, nullptr);
-    if (al_bmp)
+    if (bmp)
     {
-        _alBitmap = al_bmp;
-        _isDataOwner = true;
+        this->CreateCopy(bmp.get());

```

Comment:
This makes an completely unnecessary copy operation of the bitmap.

My earlier proposal was to remove Bitmap members LoadFromFile and SaveToFile completely, as they are no longer needed.

Optionally, you could separate a function that loads an array of pixels, and let create a Bitmap object from that array, knowing parameters.

Reply:
erh, we never use this method, so I simply deleted it.